[LISKDOCS-52] Add Remix test example (#302)
LiskHQ/lisk-docs: docs/partials/_remix-deploy-example.mdx

> 

Reviewer: Broken error found!
<img width="1057" alt="image" src="https://github.com/user-attachments/assets/144fca2b-84dc-4e6c-806c-4fbb5efde0f8" />

Author: fixed 👍

Press the green play button at the top to compile the contract.

## 5. Test the contract
To test the contract, we are using the [Solidity Unit Testing](https://remix-ide.readthedocs.io/en/latest/unittesting.html) plugin of the Remix IDE.

> If you haven't used this plugin before and are not seeing the double check icon in the sidebar, you first have to [activate](https://remix-ide.readthedocs.io/en/latest/unittesting.html) it from Remix plugin manager.

Reviewer: ```suggestion
If you haven't used this plugin before and are not seeing the double-check icon in the sidebar, you first have to [activate](https://remix-ide.readthedocs.io/en/latest/unittesting.html) it from the Remix plugin manager.
```

To start testing, create a new folder called `tests` at the root level of your Remix workspace.
This is the default directory for tests.
If you want to change the test directory, you can do so by clicking on the double-check icon and specifying the path in the `Test directory` field.

Next, open the Solidity file which you want to test, then switch to the "Solidity Unit Testing" plugin and click on the `Generate` button.

> In the test directory, a test file will be created for your selected file.

Reviewer 2: ```suggestion
This will create a test file (e.g., myNFT_test.sol) in the tests directory.
```

    function beforeAll() public {
        // <instantiate contract>
        Assert.equal(uint(1), uint(1), "1 should be equal to 1");
    }

    function checkSuccess() public {
        // Use 'Assert' methods: https://remix-ide.readthedocs.io/en/latest/assert_library.html
        Assert.ok(2 == 2, 'should be true');
        Assert.greaterThan(uint(2), uint(1), "2 should be greater than to 1");
        Assert.lesserThan(uint(2), uint(3), "2 should be lesser than to 3");
    }

    function checkSuccess2() public pure returns (bool) {
        // Use the return value (true or false) to test the contract
        return true;
    }
    
    function checkFailure() public {
        Assert.notEqual(uint(1), uint(1), "1 should not be equal to 1");
    }

    /// Custom Transaction Context: https://remix-ide.readthedocs.io/en/latest/unittesting.html#customization
    /// #sender: account-1
    /// #value: 100
    function checkSenderAndValue() public payable {
        // account index varies 0-9, value is in wei
        Assert.equal(msg.sender, TestsAccounts.getAccount(1), "Invalid sender");
        Assert.equal(msg.value, 100, "Invalid value");
    }
}
```

> Take the auto-generated template as a basis to create sufficient unit tests to ensure that your contract works as expected under different scenarios.

Reviewer 2: ```suggestion
Take the auto-generated template as a basis to create sufficient unit tests that verify your contract behavior in various scenarios.
```

Remix injects a built-in assert library for testing.

> You can visit the library documentation [here](https://remix-ide.readthedocs.io/en/latest/assert_library).

Reviewer: URL returns 404. Also, instead of **here**, please use explicit heading text. 

<img width="626" alt="image" src="https://github.com/user-attachments/assets/f2bcf018-e355-47d2-b396-9c49e24309be" />

Author: fixed 👍

        uint256 initialTokenId = nft.currentTokenId();
        
        // Mint a new NFT to this contract's address
        uint256 newTokenId = nft.mint(TestsAccounts.getAccount(2));
        
        // Check that token ID incremented
        Assert.equal(newTokenId, initialTokenId + 1, "Token ID should increment by 1");
        
        // Check currentTokenId updated
        Assert.equal(nft.currentTokenId(), newTokenId, "currentTokenId should match new token ID");
        
        // Check ownership
        Assert.equal(nft.ownerOf(newTokenId), TestsAccounts.getAccount(2), "Contract should own the minted NFT");
    }
    
    // Test minting to specific address
    function testMintToAddress() public {
        
        // Mint NFT to test address
        uint256 newTokenId = nft.mint(TestsAccounts.getAccount(1));
        
        // Check ownership
        Assert.equal(nft.ownerOf(newTokenId), TestsAccounts.getAccount(1), "Test address should own the minted NFT");
        
        // Check token balance
        Assert.equal(nft.balanceOf(TestsAccounts.getAccount(1)), 1, "Test address should have 1 NFT");
    }
}
```

There are several ways to customize your solidity tests.

> For example you can [customize the compiler and the transaction context](https://remix-ide.readthedocs.io/en/latest/unittesting.html#customization).

Reviewer: ```suggestion
For example, you can [customize the compiler and the transaction context](https://remix-ide.readthedocs.io/en/latest/unittesting.html#customization).
```

Once you are done with writing tests, select the test file(s) and click on `Run` to execute the tests.
The execution will run in a separate environment.
After completing the execution of one file, a test summary will be shown:

```text

> Result for tests/MyNFT_test.sol

Reviewer 2: Test file name is not consistent.
```suggestion
Result for tests/myNFT_test.sol
```

> Result for tests/MyNFT_test.sol
> Passed: 3
> Failed: 0
> Time Taken: 0.05s

Reviewer: The `Passed` value should be `2` since we have created two test functions.
```suggestion
Result for tests/MyNFT_test.sol
Passed: 2
Failed: 0
Time Taken: 0.06s
```
